Sources/Swift/Protocol/Metric.swift

```diff
@@ -0,0 +1,289 @@
+    ///
+    /// Attributes provide additional context and can be used for filtering and
+    /// grouping metrics in Sentry. Common attributes include endpoint names,
+    /// HTTP methods, status codes, etc.
+    public var attributes: [String: SentryAttribute]
+
+    /// Creates a metric entry with the specified properties.
+    ///
+    /// - Note: This initializer is internal. Metrics should be created by the SDK through the public metrics API.
+    ///         Users can modify metrics in the `beforeSendMetric` callback.
+    ///
+    /// - Parameters:
+    ///   - timestamp: The timestamp when the metric was recorded
+    ///   - traceId: The trace ID to associate this metric with distributed tracing
+    ///   - name: The name of the metric
+    ///   - value: The numeric value of the metric
+    ///   - metricType: The type of metric
+    ///   - unit: The unit of measurement for the metric value (optional)
+    ///   - attributes: A dictionary of structured attributes to add to the metric
+    internal init(
+        timestamp: Date,
+        traceId: SentryId,
+        name: String,
+        value: MetricValue,
+        type: MetricType,
+        unit: String?,
+        attributes: [String: SentryAttribute]
+    ) {
+        self.timestamp = timestamp
+        self.traceId = traceId
+        self.name = name
+        self.metricType = type
+        self.unit = unit
+        self.attributes = attributes
+        
+        // Perform type conversion when needed
+        switch (type, value) {
+        case (.counter, .integer), (.gauge, .double), (.distribution, .double):
+            self._value = value
+        case (.counter, .double(let doubleValue)):
+            // Convert double to integer by flooring
+            SentrySDKLog.warning("Counter metric created with double value \(doubleValue). Converting to integer by flooring: \(Int64(floor(doubleValue))).")
+            self._value = .integer(Int64(floor(doubleValue)))
+        case (.gauge, .integer(let intValue)):
+            // Convert integer to double
+            SentrySDKLog.warning("Gauge metric created with integer value \(intValue). Converting to double: \(Double(intValue)).")
+            self._value = .double(Double(intValue))
+        case (.distribution, .integer(let intValue)):
+            // Convert integer to double
+            SentrySDKLog.warning("Distribution metric created with integer value \(intValue). Converting to double: \(Double(intValue)).")
+            self._value = .double(Double(intValue))
+        }
+    }
+    
+    /// Adds or updates an attribute in the metric entry.
+    /// - Parameters:
+    ///   - attribute: The attribute value to add
+    ///   - key: The key for the attribute
+    public mutating func setAttribute(_ attribute: SentryAttribute?, forKey key: String) {
+        if let attribute = attribute {
+            attributes[key] = attribute
+        } else {
+            attributes.removeValue(forKey: key)
+        }
```

Comment:
`l`: A bit shorter, if I'm not mistaken.
```suggestion
        attributes[key] = attribute
```